### PR TITLE
Add the ability to enrich the CloudWatch events

### PR DESCRIPTION
**Pull request by joaoasrosa**
This PR adds the ability to enrich the CloudWatch events at ship time. It addresses issue #23.
If the property is set in the correct format (`key1=value1;key2=value2`) the Lambda function will enrich the log with the properties:
```
... LOG EVENT ...
key1: "value1",
key2: "value2"
```
It will allow enriching the log with custom properties, thus querying on Logz.io will be easier.

### CODE REVIEW

File: cloudwatch/src/lambda_function.py [hunk 1]
```diff
@@ -76,6 +76,16 @@ def _parse_cloudwatch_log(log, aws_logs_data, log_type):
...
+    # If ENRICH has value, add the properties
+    try:
+        if os.environ['ENRICH']:
+            properties_to_enrich = os.environ['ENRICH'].split(";")
+            for property_to_enrich in properties_to_enrich:
+                property_key_value = property_to_enrich.split("=")
+                log[property_key_value[0]] = property_key_value[1]
+    except (KeyError, ValueError):
+        pass
+
```
Comment (idohalevi) [inline cloudwatch/src/lambda_function.py]: Lets put this in `_enrich_logs_data`, that way we can do it once in an event and not every log line

File: cloudwatch/tests/lambda_tests.py [hunk 2]
```diff
@@ -267,6 +267,25 @@ def test_large_body(self):
...
+            self.fail("Failed on handling a legit event. Expected status_code = 200")
+
+        request = httpretty.HTTPretty.last_request
+        body_logs_list = request.body.splitlines()
+
+        for i in xrange(BODY_SIZE):
+            json_body_log = json.loads(body_logs_list[i])
+            self.assertEqual(json_body_log['environment'], "testing")
+            self.assertEqual(json_body_log['foo'], "bar")
+
```
Comment (idohalevi) [inline cloudwatch/tests/lambda_tests.py]: Looking good, I think we are missing some edge cases. Let's add tests for an empty string and bad format one.

**Comment by joaoasrosa** [comment]
@idohalevi thanks for the review. All the comments were addressed. 
Let me know if I miss any spot.

File: cloudwatch/src/lambda_function.py [hunk 3]
```diff
@@ -67,6 +67,10 @@ def _parse_cloudwatch_log(log, aws_logs_data, log_type):
     log['invoked_function_arn'] = aws_logs_data['invoked_function_arn']
     log['type'] = log_type
 
+    if 'data_to_enrich' in aws_logs_data:
```
Comment (idohalevi) [inline cloudwatch/src/lambda_function.py]: Once I saw your PR I thought of a better way to implement it:
Instead of log['x'] = aws_logs_data['y']
We can do:
```suggestion
 log.update(aws_logs_data)
```

File: cloudwatch/src/lambda_function.py [hunk 4]
```diff
@@ -82,6 +86,15 @@ def _enrich_logs_data(aws_logs_data, context):
...
         aws_logs_data['function_version'] = context.function_version
         aws_logs_data['invoked_function_arn'] = context.invoked_function_arn
+
+        # If ENRICH has value, add the properties
+        if os.environ['ENRICH']:
+            data_to_enrich = dict()
+            properties_to_enrich = os.environ['ENRICH'].split(";")
+            for property_to_enrich in properties_to_enrich:
+                property_key_value = property_to_enrich.split("=")
+                data_to_enrich[property_key_value[0]] = property_key_value[1]
```
Comment (idohalevi) [inline cloudwatch/src/lambda_function.py]: And here you can just do:
```suggestion
               aws_logs_data[property_key_value[0]] = property_key_value[1]
```

**Comment by joaoasrosa** [comment]
@idohalevi something strange happens with the suggestions, and given my limit knowledge of Python, I do not know what's going on. As you can see in the separate branch (`https://github.com/joaoasrosa/logzio_aws_serverless/commit/07595704e99cc89a20bfe951a94a2de4428dd980`), I'd applied the changes. However, all the tests fail. Any idea?

**Comment by idohalevi** [comment]
@joaoasrosa I found the bug and the tests work. I'll close this PR and push a new one with the fixes if it's ok by you.

**Comment by idohalevi** [comment]
Done here #25
